controllers/user_controller.go

```diff
@@ -0,0 +1,428 @@
+		return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "S3UserCreationFailed",
+			fmt.Sprintf("Creation of user [%s] on S3 instance has failed", userResource.Spec.Name), err)
+	}
+
+	policies := userResource.Spec.Policies
+	if policies != nil {
+		err := r.S3Client.AddPoliciesToUser(username, policies)
+		if err != nil {
+			logger.Error(err, "an error occurred while adding policy to user", "user", userResource.Spec.Name)
+			return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "S3UserCreationFailed",
+				fmt.Sprintf("Error while updating policies of user [%s] on S3 instance has failed", userResource.Spec.Name), err)
+		}
+
+	}
+
+	groups := userResource.Spec.Groups
+	if groups != nil {
+		err := r.S3Client.AddGroupsToUser(username, groups)
+		if err != nil {
+			logger.Error(err, "An error occurred while adding groups to user", "user", userResource.Spec.Name)
+			return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "S3UserCreationFailed",
+				fmt.Sprintf("Error while updating groups of user [%s] on S3 instance has failed", userResource.Spec.Name), err)
+		}
+	}
+
+	// Define a new K8S Secrets
+	data := make(map[string][]byte)
+	data["accessKey"] = []byte(username)
+	data["secretKey"] = []byte(password)
+	secret, err := newSecretForCR(userResource, data)
+
+	if err != nil {
+		// Error while creating the Kubernetes secret - requeue the request.
+		logger.Error(err, "Could not generate Kubernetes secret")
+		return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "SecretGenerationFailed",
+			fmt.Sprintf("Generation of k8s secrets [%s] has failed", userResource.Spec.Name), err)
+	}
+
+	// Set S3User instance as the owner and controller
+	err = ctrl.SetControllerReference(userResource, secret, r.Scheme)
+	if err != nil {
+		logger.Error(err, "Could not set owner of kubernetes secret")
+		return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "SecretGenerationFailed",
+			fmt.Sprintf("Generation of k8s secrets [%s] has failed", userResource.Spec.Name), err)
+	}
+
+	// Check if this Secret already exists
+	err = r.Get(ctx, types.NamespacedName{Name: secret.Name, Namespace: secret.Namespace}, &corev1.Secret{})
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Creating a new Secret", "Secret.Namespace", secret.Namespace, "Secret.Name", secret.Name)
+		err = r.Create(ctx, secret)
+		if err != nil {
+			logger.Error(err, "Could not create secret")
+			return r.setS3UserStatusConditionAndUpdate(ctx, userResource, "OperatorFailed", metav1.ConditionFalse, "SecretCreationFailed",
+				fmt.Sprintf("Generation of k8s secrets [%s] has failed", secret.Name), err)
+		}
+		// Secret created successfully - requeue only if no version is specified
```

Comment:
Not sure what this requeue is about ? Actually, I don't see the requeue at all.

Reply:
it's a rest of a simple copy paste 🤡